zio-profiling/jvm/src/main/scala/zio/profiling/causal/ScopeSelector.scala

```diff
@@ -14,18 +14,23 @@
  * limitations under the License.
  */
 
-package zio.profiling
+package zio.profiling.causal
 
-import scala.reflect.ClassTag
+import zio.profiling.Tag
 
-final private class VolatileArray[A: ClassTag](val length: Int) {
+final case class ScopeSelector(run: Tag => Option[Tag])
 
-  private[this] val array = new Array[A](length)
+object ScopeSelector {
```

Comment:
The idea here is being able to select the candidates for optimization:

`Root / foo / *` -> consider all tags or locations directly nested in the foo tag for optimization

`Root / foo / **` -> consider all 'leave' locations nested in the foo tag for optimization